Apply v-pre to all anchor elements in pageNav

tlylt opened the pull request:
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [x] Code maintenance
- [ ] Others, please explain:

<!--
  If this pull request is addressing an issue, link to the issue: "Fixes #xxx" or "Resolves #xxx"

  If the pull request completely addresses the issue, use one of the issue closing keywords. https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords

  Otherwise, elaborate further on the rationale of this pull request as needed
-->

**Overview of changes:**
Based on the comment: https://github.com/MarkBind/markbind/pull/1724#issuecomment-1017376544
Previously:
- v-pre is added to <a> tag on a need basis, i.e. there are `{}` in the header text of pageNav headers, and therefore keep to escape from Vue rendering.

Now:
- v-pre is added to all <a> tag for pageNav headers, for simplicity and performance sake.

Functional test files have been updated to deal with the addition of v-pre in HTML and in vue render files:
- ![pr1](https://user-images.githubusercontent.com/41845017/150463218-cf6b0b11-79e2-4855-b55b-463deb4b8ef1.PNG)
- ![pr2](https://user-images.githubusercontent.com/41845017/150463206-54768523-c9e3-47f7-9e10-e8fcdce4369b.PNG)


**Anything you'd like to highlight / discuss:**
nil

**Testing instructions:**
nil

<!--
  Any special testing instructions **not including** our automated tests.
-->

**Proposed commit message: (wrap lines at 72 characters)**
Add v-pre to all anchor elements in pageNav

v-pre is added to anchor elements of pageNav on a need basis
to escape raw {} syntax.

Let's instead add v-pre to all pageNav anchor elements to
keep it simple and performant.

<!--
  See this link for more info on how to write a good commit message:
  https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message

  As best as possible, write a succinct commit title in 50 characters
-->

---

**Checklist:** :ballot_box_with_check:

<!-- Leave non-applicable items unchecked -->

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [ ] Linked all related issues
- [x] No unrelated changes <!-- Its tempting, but increases the reviewer's work, and really pollutes the commit history =( -->

<!--
  We'll try our best to get to your PR within a week.
  If we haven't gotten to it then, or if your pull request resolves an urgent item, feel free to give us a ping!
-->

jonahtanjz reviewed (commented):
Thanks for making the change @tlylt :)

I also noticed now that the page nav title has the same issue.

In the `frontmatter`, if there are curly braces used in the pageNavTitle similar to the headers,
```html
<frontmatter>
  layout: default.md
  pageNav: 2
  pageNavTitle: {% raw %} "Title interpolation {{ test }}" {% endraw %}
</frontmatter>
```
it will result in the same issue.

Will have to add `v-pre` to [page nav title anchor](https://github.com/MarkBind/markbind/blob/9464ab87f6345b9fef33c861c8f37747c613917e/packages/core/src/Page/index.js#L410) as well.

tlylt commented:
> In the `frontmatter`, if there are curly braces used in the pageNavTitle similar to the headers,
> 
> ```
> <frontmatter>
>   layout: default.md
>   pageNav: 2
>   pageNavTitle: {% raw %} "Title interpolation {{ test }}" {% endraw %}
> </frontmatter>
> ```
> 
> it will result in the same issue.

Thanks @jonahtanjz! I have addressed the issue and added a test case for it.

jonahtanjz reviewed (approved):
Thanks @tlylt!

LGTM 👍
